src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs

```diff
@@ -130,7 +137,7 @@ private IBlockImprovementContext CreateBlockImprovementContext(string payloadId,
                 {
                     if (_logger.IsTrace) _logger.Trace($"Block for payload {payloadId} with parent {parentHeader.ToString(BlockHeader.Format.FullHashAndNumber)} will be improved in {_improvementDelay.TotalMilliseconds}ms");
                     await Task.Delay(_improvementDelay);
-                    if (!blockImprovementContext.Disposed) // if GetPayload wasn't called for this item or it wasn't cleared
+                    if (!blockImprovementContext.Disposed && t.IsCanceled == false) // if GetPayload wasn't called for this item or it wasn't cleared
```

Comment:
Could be faulted as well as cancelled?
```suggestion
                    if (!blockImprovementContext.Disposed && t.IsCompletedSuccessfully) // if GetPayload wasn't called for this item or it wasn't cleared
```

Reply:
Yup, I was thinking about it. Unfortunately, the current fix doesn't resolve the issue. I confirmed with `Cannot_produce_bad_blocks`